testsuite/testclient/conftest.py

```diff
@@ -0,0 +1,900 @@
+"""
+Module for running the testclient .yaml test cases.
+"""
+
+import doctest
+import socket
+import pytest
+import re
+import traceback
+import threading
+from collections import namedtuple
+try:
+    from collections import OrderedDict
+except ImportError:
+    from ordereddict import OrderedDict
+import yaml
+import yamlordereddictloader
+import httpretty
+from httpretty.core import HTTPrettyRequestEmpty, fakesock
+from lxml import etree, doctestcompare
+import six
+import pywbem
+from pywbem._utils import _ensure_unicode
+from pywbem._nocasedict import NocaseDict
+
+
+class ExcThread(threading.Thread):
+    """
+    Thread that catches an exception and passes it to the creator
+    in the Thread.join() method.
+    """
+
+    def run(self):
+        self.exc = None
+        try:
+            threading.Thread.run(self)
+        except Exception:
+            import sys
+            self.exc = sys.exc_info()
+
+    def join(self, *args, **kwargs):
+        threading.Thread.join(self, *args, **kwargs)
+        if self.exc:
+            six.reraise(*self.exc)
+
+
+def patched_makefile(self, mode='r', bufsize=-1):
```

Comment:
Can we rename.  Took quite a while to sort out what makefile meant since the work Makefile has a very pecular usage.

Reply:
I added a description on what it means, ands hope we can live with that. The original function is called makefile() and I think we should keep the connection, naming-wise.
DONE.

Reply:
I am happy.  That one threw me until I dug far enought into pytest doc